### PR TITLE
Move mirror to a standalone package from models

### PR DESCRIPTION
**Pull request by lunny**
This PR will move mirrors codes unrelated with database from `models` to a standalone package `services/mirror`. But when import it could be aliased `mirror_service "code.gitea.io/gitea/services/mirror"`

### CODE REVIEW

File: models/repo.go [hunk 1]
```diff
@@ -1699,6 +1699,12 @@ func UpdateRepository(repo *Repository, visibilityChanged bool) (err error) {
 	return sess.Commit()
 }
 
+// UpdateRepositoryUpdatedTime updates a repository's updated time
+func UpdateRepositoryUpdatedTime(repoID int64, updateTime time.Time) error {
```
Comment (kolaente) [inline models/repo.go:1724]: Where does this function come from?

Comment (lunny) [reply models/repo.go:1724]: Come from https://github.com/go-gitea/gitea/pull/7486/files#diff-6b299323ba0a072129c7841649f49536L431

Comment (kolaente) [reply models/repo.go:1724]: Ah ok, fair enough.

**Comment by lunny** [comment]
Ready to review.

**Review by guillep2k — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@guillep2k done.

**Review by guillep2k — APPROVED** [review]

**Comment by guillep2k** [comment]
You really are out of luck with this PR. 😄 
image

**Review by sapk — APPROVED** [review]

**Pull request merged**